src/components/select/utils/with-filter.hoc.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import I18n from 'i18n-js';
+import filterChildren from '../../../utils/filter-children';
+import StyledOption from '../option/option.style';
+import highlightPartOfText from './highlight-part-of-text';
+
+/** Filters wrapped component children based on provided filter text and highlights matching content */
+const withFilter = (WrappedComponent) => {
+  const FilteredComponent = ({
+    children,
+    filterText,
+    forwardRef,
+    ...props
+  }) => {
+    function getFilteredChildren() {
+      let filteredElements = children;
+
+      if (filterText) {
+        filteredElements = filterChildren({ value: filterText })(children);
+
+        if (!filteredElements) {
+          const noResultsText = I18n.t('select.no_results', {
+            defaultValue: 'No results'
```

Comment:
We should have a prop for this if we're going to support `placeholder`.

Reply:
a placeholder in the option list?

Reply:
At the moment we have to use I18n to change the "No results" but we can change the placeholder by using a prop or I18n. We should be able to change the "No results" text by using a prop too.

Reply:
Unresolved.

Reply:
added a prop with default value